### PR TITLE
AG-12190 - Rework setLicenseKey() API to match Grid.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-12190

### CODE REVIEW

**Review by taktran — APPROVED** [review]
LGTM 👍🏼

File: packages/ag-charts-enterprise/src/main-modules.ts [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+import { LicenseManager as RealLicenseManager } from './license/licenseManager';
 import { setupEnterpriseModules as internalSetup } from './setup';
 
-export { AgCharts, VERSION };
+const LicenseManager = {
+    setLicenseKey(key: string) {
+        RealLicenseManager.setLicenseKey(key);
+    },
+};
+export { AgCharts, VERSION, LicenseManager };
```
Comment (iMoses) [inline packages/ag-charts-enterprise/src/main-modules.ts:12]: why aren't we exporting the `RealLicenseManager`?

File: packages/ag-charts-types/src/integratedCharts.ts [hunk 2]
```diff
@@ -97,7 +97,7 @@ export interface IntegratedModule {
     create(options: AgChartInstanceOptions): AgChartInstance<AgChartInstanceOptions>;
     createSparkline(options: AgSparklineOptions): AgChartInstance<AgSparklineOptions>;
     setup(): void;
-    setGridContext(gridContext: boolean): void;
-    setLicenseKey(licenseKey: string): void;
+    setGridContext?(gridContext: boolean): void;
+    setLicenseKey?(licenseKey: string): void;
```
Comment (iMoses) [inline packages/ag-charts-types/src/integratedCharts.ts:101]: It was Stephen who asked for a consistent API between community and enterprise.
I personally think it's fine, but maybe worth checking that it's consistent with the grid's approach.

Comment (alantreadway) [reply packages/ag-charts-types/src/integratedCharts.ts:101]: https://github.com/ag-grid/ag-grid/pull/9292/files

File: packages/ag-charts-enterprise/src/main-modules.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-enterprise/src/main-modules.ts:12]: Just from the perspective of only exporting the absolutely necessary API, `LicenseManager` has a larger API surface than users need to consider, so this 1) simplifies the exported API (from a user perspective) and 2) avoids users being able to do unexpected things with the `LicenseManager`.

File: packages/ag-charts-types/src/integratedCharts.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-types/src/integratedCharts.ts:101]: I've requested review from @StephenCooper

**Pull request merged**